lib/Elastica/Transport/Guzzle.php

```diff
@@ -156,7 +175,12 @@ protected function _getBaseUrl(Connection $connection)
         if (!empty($url)) {
             $baseUri = $url;
         } else {
-            $baseUri = $this->_scheme.'://'.$connection->getHost().':'.$connection->getPort().'/'.$connection->getPath();
+            $baseUri = (string) Uri::fromParts([
+                'scheme' => $this->_scheme,
+                'host' => $connection->getHost(),
+                'port' => $connection->getPort(),
+                'path' => ltrim('/', $connection->getPath()),
```

Comment:
I didn't realise that the params were turned around. Definitively doesn't make sense. 

Did you stumble over this because it broke something? If no, it seems to just work and we can remove it? Otherwise, we should provide a fix.